boards/FSM/Inc/App/App_FlowMeterSignals.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <stdbool.h>
+#include "App_FsmWorld.h"
+
+/**
+ * Check if the primary flow rate is below its minimum threshold
+ * @param world The world to get the primary flow rate in-range check
+ * @return true if the primary flow rate (L/min) is below its minimum flow rate
+ * threshold, else false
+ */
+bool App_FlowMetersSignals_IsPrimaryFlowRateBelowThreshold(
+    struct FsmWorld *world);
+
+/**
+ * Check if the primary flow rate is in range
+ * @param world The world to get the primary flow rate in-range check
+ * @return true if the primary flow rate (L/min) is in range, else false
+ */
+bool App_FlowMetersSignals_IsPrimaryFlowRateInRange(struct FsmWorld *world);
+
+/**
+ * Check if the secondary flow rate is below its minimum threshold
+ * @param world The world to get the secondary flow rate in-range check
+ * @return true if the secondary flow rate (L/min) is in range, else false
+ */
+bool App_FlowMetersSignals_IsSecondaryFlowRateBelowThreshold(
+    struct FsmWorld *world);
+
+/**
+ * Check if the secondary flow rate is in range
+ * @param world The world to get the secondary flow rate in-range check
+ * @return true if the secondary flow rate (L/min) is in range
+ */
+bool App_FlowMetersSignals_IsSecondaryFlowRateInRange(struct FsmWorld *world);
+
+/**
+ * The callback function for the primary flow rate signal in the given world
```

Comment:
Since these are called in the case of underflow, I think they should be named something like PrimaryFlowRateBelowThresholdCallback

Reply:
👍👍